iina/Extensions.swift

```diff
@@ -630,6 +630,22 @@ extension NSScreen {
       Logger.log("\(label): visible frame \(screen.visibleFrame)")
     }
   }
+
+  /// Set the frame of the given window and its content view to use all the usable area of this screen.
+  /// - parameter window: The window to set the frame of.
+  /// - parameter displayFlag: Specifies whether the window redraws the views that need to be displayed. When true the
+  ///   window sends a [displayIfNeeded()](https://developer.apple.com/documentation/appkit/nswindow/1419096-displayifneeded)
+  ///   message down its view hierarchy, thus redrawing all views.
+  /// - parameter animateFlag: Specifies whether the window performs a smooth resize. true to perform the animation, whose
+  ///   duration is specified by [animationResizeTime(_:)](https://developer.apple.com/documentation/appkit/nswindow/1419655-animationresizetime)
+  func setFrame(_ window: NSWindow, display displayFlag: Bool, animate animateFlag: Bool) {
+    window.setFrame(frame, display: displayFlag, animate: animateFlag)
+    guard let unusable = cameraHousingHeight  else { return }
+    // This screen contains an embedded camera. Shorten the height of the window's content view's
+    // frame to avoid having part of the window obscured by the camera housing.
+    let view = window.contentView!
+    view.setFrameSize(NSMakeSize(view.frame.width, frame.height - unusable))
+  }
```

Comment:
Here are my thoughts:

- It's redundant to have both `screen` and `window`. The screen should always be `window.screen ?? NSScree.main`; it makes little sense to call this method on a screen different from where the window is.
- We don't see much potential to use this method somewhere else. This shouldn't be a general method; Apart from setting the window frame, it also changes the content view, which is not expected for a method named `setFrame`.
- So, since we only use it in this specific case (legacy screen), it's better to rename the method to reflect the actual use and remove the argument. (Do we need to override `display` or `animate` in the future? If not, we can remove them.)
- If we expect the target window to be the `MainWindow`, it's better to move it to `MainWindowController`.

Therefore, I suggest putting it in `MainWindowController` with name
```swift
setWindowFrameForLegacyFullScreen()
```

Alternatively, if write as an extension, it should be an `NSWindow` extension, and its name should reflect the purpose:
```swift
window.setFrameAndContentViewToFullUseableSize()
```